### PR TITLE
Assert http2 header ordering

### PR DESCRIPTION
**Pull request by dgquintas**

### CODE REVIEW

File: src/core/transport/chttp2/stream_encoder.c [hunk 1]
```diff
@@ -361,6 +362,16 @@ static grpc_mdelem *hpack_enc(grpc_chttp2_hpack_compressor *c,
   gpr_uint32 indices_key;
   int should_add_elem;
 
+  if (GPR_SLICE_LENGTH(elem->key->slice) > 0) {
```
Comment (dgquintas) [inline src/core/transport/chttp2/stream_encoder.c]: maybe this could be an assert? are we expecting keys to always be non-empty?

Comment (ctiller) [inline src/core/transport/chttp2/stream_encoder.c]: I'd be happy with an assert here.

Comment (dgquintas) [inline src/core/transport/chttp2/stream_encoder.c]: done

**Comment by dgquintas** [comment]
greendom!

**Pull request merged**